packages/fork-choice/src/forkChoice/forkChoice.ts

```diff
@@ -223,6 +224,52 @@ export class ForkChoice implements IForkChoice {
     }
   }
 
+  // Basically called by `predictProposerHead`. If the result is not same as blockRoot's block, return true else false
+  // See https://github.com/ethereum/consensus-specs/blob/a94d85bb3b33f84a5830eb98945b790b02ae80e8/specs/bellatrix/fork-choice.md#should_override_forkchoice_update
+  // Return true if the given block passes all criteria to be re-orged out
+  // Return false otherwise.
+  // Note when proposer boost reorg is disabled, it always returns false
+  shouldOverrideForkChoiceUpdate(currentSlot: Slot, blockRoot: RootHex): ShouldOverrideForkChoiceUpdateResult {
+    const headBlock = this.getBlockHex(blockRoot);
+    if (headBlock === null) {
+      // should not happen beacause this block just got imported. Fall back to no-reorg.
+      return {shouldOverrideFcu: false, reason: NotReorgedReason.HeadBlockNotAvailable};
+    }
+    // Skip re-org attempt if proposer boost (reorg) are disabled
+    if (!this.opts?.proposerBoost || !this.opts?.proposerBoostReorg) {
+      this.logger?.verbose("Skip shouldOverrideForkChoiceUpdate check since the related flags are disabled", {
+        slot: currentSlot,
+      });
```

Comment:
could print out flags here

```suggestion
      this.logger?.verbose("Skip shouldOverrideForkChoiceUpdate check since the related flags are disabled", {
        slot: currentSlot,
        proposerBoost: this.opts?.proposerBoost,
        proposerBoostReorg: this.opts?.proposerBoostReorg,
      });
```

could destructure opts above if-statement